### DISCUSSION TITLE
Can I start with Carbon without any programming knowledge?

### DISCUSSION THREAD

**Discussion by tanviruli03** [carbon-language/carbon-lang#1551] [Getting started]
Hey dear friends or  honorable mentors!

I don't have any knowledge on other programing languages as well as C++. Can I start my programming journey with new Carbon language? Is it right choice for me to start my journey? 

Thanks

**Comment by EdwardStables** [comment]
Carbon is at a very early stage of its life, with many features currently unconfirmed or liable to change. I would not suggest it as your first programming language at all.

**Reply by jonmeow** [reply]
For reference: https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/faq.md#how-soon-can-we-use-carbon

**Reply by RylanYancey** [reply]
While it's still not a great first language, Rust is also a fairly new language but is much more mature and stable. It's also blazingly fast 🚀 🚀 🚀 🚀

**Reply by yrsdi** [reply]
carbon is still experimental. .suggested to trying Rust-lang, zig-lang, or nim-lang ;)

**Comment by tszdanger** [comment]
It would not be a nice choice as choosing carbon as your first PL.

**Comment by YagaoDirac** [comment]
My answer is, the basic part of python, and then the basic part of js, and see if you are interested in c++. 
Then data structure, algorithm, sortation.
Then c++ std library.
Then some more project experience.
Then if Carbon is good enough at that time, it would be worthy your time. 
Carbon is probably going to a super crazy place where tons a knowledge are needed to use it.

**Comment by mshahrin** [comment]
Start learning one programming language. Once you have the hang of it/learn bits of it, move on to learning about object orient programming. You can choose any language you want but just not Carbon for now. I started learning C++ as my first language in college, you can see what suits you best. Many people had said that they found Python PL to be an easier option. You choose which one you want. However, you absolutely need the knowledge of programming language for the future.

**Comment by prathmesh-b** [comment]
Never, this is still in beta state. Instead go for c/c++/java. I am currently learning java because of oop. you should also learn from other stable languages.
